### PR TITLE
Standard migration command

### LINKED ISSUE #2197: Run tenant migrations with a command

### PR DESCRIPTION
**Pull request by Weiko**
## Context

Fixes #2197 

We want to be able to run migrations on existing tenants. To do that we introduced a command runner and 2 commands.
- tenant:migration will insert the new standard migrations inside the tenant_migrations table and execute them
- tenant:sync-metadata will insert the objects and fields to match our standard objects. Currently it's deleting everything and recreating them but we should be able to do a merge there.

Note: The commands require a workspace_id. In the end, we want to be able to omit this parameter and let the command fetch all the existing workspaces. We will need to handle properly migrations that fail in the middle of the execution and be able to run it multiple times if necessary and the command should run idempotent jobs.

### CODE REVIEW

File: server/package.json [hunk 1]
```diff
@@ -34,7 +34,8 @@
     "database:migrate": "yarn typeorm:migrate && yarn prisma:migrate",
     "database:generate": "yarn prisma:generate",
     "database:seed": "yarn prisma:seed",
-    "database:reset": "yarn database:generate && yarn database:migrate && yarn database:seed"
+    "database:reset": "yarn database:generate && yarn database:migrate && yarn database:seed",
+    "command": "node dist/src/command"
```
Comment (Weiko) [inline server/package.json:38]: There is probably a better name

File: server/src/metadata/commands/sync-tenant-metadata.command.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    // TODO: run in a dedicated job + run queries in a transaction.
+    const dataSourceMetadata =
+      await this.dataSourceMetadataService.getLastDataSourceMetadataFromWorkspaceIdOrFail(
+        options.workspaceId,
+      );
+
+    // TODO: This solution could be improved, using a diff for example, we should not have to delete all metadata and recreate them.
+    await this.fieldMetadataService.deleteMany(
+      { workspaceId: { eq: options.workspaceId } },
+      { useSoftDelete: false },
```
Comment (Weiko) [inline server/src/metadata/commands/sync-tenant-metadata.command.ts:40]: This useSoftDelete does not work as I expected. It's still softDeleting our objects and we don't want that here... I had to temporarily deactivate this default behaviour inside fieldMetadataService and objectMetadataService

File: server/src/metadata/tenant-initialisation/tenant-initialisation.service.ts [hunk 3]
```diff
@@ -67,7 +67,7 @@ export class TenantInitialisationService {
    * @param dataSourceMetadataId
    * @param workspaceId
    */
-  private async createObjectsAndFieldsMetadata(
+  public async createObjectsAndFieldsMetadata(
```
Comment (Weiko) [inline server/src/metadata/tenant-initialisation/tenant-initialisation.service.ts:70]: As mentioned in the PR, we don't want to call TenantInitialisation for migrations, we should move this responsibility somewhere else.

File: server/package.json (hunk 1 above)
Comment (magrinj) [reply server/package.json:38]: @Weiko Yes, we should put something like `twenty:migrate` or `tenant:migrate`

File: server/src/metadata/commands/run-tenant-migrations.command.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+// TODO: implement dry-run
+interface RunTenantMigrationsOptions {
+  workspaceId: string;
+}
+
+@Command({
+  name: 'tenant:migrate',
+  description: 'Run tenant migrations',
+})
+export class RunTenantMigrations extends CommandRunner {
```
Comment (magrinj) [inline server/src/metadata/commands/run-tenant-migrations.command.ts:15]: @Weiko The mismatch between the CommandRunner class and the CommandModule is a bit strange, we should align them I guess:
`TenantMigrationCommandModule`
`TenantMigrationCommandRunner`

File: server/src/metadata/commands/sync-tenant-metadata.command.ts (hunk 2 above)
Comment (magrinj) [reply server/src/metadata/commands/sync-tenant-metadata.command.ts:40]: @Weiko As you disabled `softDelete` is it still needed ?

File: server/src/metadata/tenant-initialisation/tenant-initialisation.service.ts (hunk 3 above)
Comment (magrinj) [reply server/src/metadata/tenant-initialisation/tenant-initialisation.service.ts:70]: I agree with that

File: server/src/metadata/commands/sync-tenant-metadata.command.ts (hunk 2 above)
Comment (Weiko) [reply server/src/metadata/commands/sync-tenant-metadata.command.ts:40]: It's not @magrinj good point, I'll remove that 🙂

File: server/package.json (hunk 1 above)
Comment (Weiko) [reply server/package.json:38]: @magrinj this one is to call the "command runner" app so you will do
`yarn command tenant:sync-metadata -w "2ab71a0e-7ced-495b-83df-d2197a6fdec1"` 
or
`yarn command tenant:migrate -w "2ab71a0e-7ced-495b-83df-d2197a6fdec1"` 
for example

File: server/src/metadata/commands/run-tenant-migrations.command.ts (hunk 4 above)
Comment (Weiko) [reply server/src/metadata/commands/run-tenant-migrations.command.ts:15]: @magrinj I'm not sure I understand.
RunTenantMigrations and SyncTenantMetadata classes are both command classes extending CommandRunner and are both provided by the same module MetadataCommandModule. Do you mean I should have 1 module per command?

**Pull request merged**